### PR TITLE
chore: update `fuel-core` to `0.38.0`

### PR DESCRIPTION
**Pull request by Dhaiwat10**
# Release notes

In this release, we:

- Updated `fuel-core` to `0.38.0`

# Summary

This PR bumps the `fuel-core` version to `0.38.0`.

# Checklist

- [x] All **changes** are **covered** by **tests** (or not applicable)
- [x] All **changes** are **documented** (or not applicable)
- [x] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [x] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: packages/fuel-gauge/src/funding-transaction.test.ts [hunk 1]
```diff
@@ -511,7 +511,7 @@ describe('Funding Transactions', () => {
       () => fundedWallet.transfer(receiver.address, transferAmount, provider.getBaseAssetId()),
       new FuelError(
         FuelError.CODES.INVALID_REQUEST,
-        'Transaction is not inserted. Hash is already known'
+        'Transaction input validation failed: Transaction id already exists (id: bacb0b8395b11e8d9a802f38674c5e41bd065567281d8c2f8f791e7760643e6a)'
```
Comment (petertonysmith94) [inline packages/fuel-gauge/src/funding-transaction.test.ts:514]:
```suggestion
    }).rejects.toThrowError(/Transaction input validation failed: UTXO \(id: .*\) does not exist/);
```

Relates to changes [here](https://github.com/FuelLabs/fuel-core/pull/2216/files#diff-763e9a79947aa33db6000118408d6f1d8e639be46a2d6ff54939f097e7986aedR28) and [here](https://github.com/FuelLabs/fuel-core/blob/94fe64e876bbb651311f1e05a6ad77c54366ee65/crates/services/txpool_v2/src/error.rs#L130-L131)

File: apps/docs-snippets/src/guide/cookbook/resubmitting-failed-transactions.test.ts [hunk 2]
```diff
@@ -70,8 +70,11 @@ describe('Resubmitting Failed Transactions', () => {
           await wallet.sendTransaction(transactionRequest);
         }
       }
+
       // #endregion resubmitting-failed-transactions-3
-    }).rejects.toThrow('Transaction is not inserted. UTXO does not exist: 0x');
+    }).rejects.toThrow(
+      'Transaction input validation failed: UTXO (id: b2814e068053151c158a8c22c219d25ba5324d1ae6efe95fc82f06fcf928527b0000) does not exist'
+    );
```
Comment (petertonysmith94) [inline apps/docs-snippets/src/guide/cookbook/resubmitting-failed-transactions.test.ts:77]:
```suggestion
    }).rejects.toThrowError(/Transaction input validation failed: UTXO \(id: .*\) does not exist/);
```

Relates to changes [here](https://github.com/FuelLabs/fuel-core/pull/2216/files#diff-763e9a79947aa33db6000118408d6f1d8e639be46a2d6ff54939f097e7986aedR28) and [here](https://github.com/FuelLabs/fuel-core/blob/94fe64e876bbb651311f1e05a6ad77c54366ee65/crates/services/txpool_v2/src/error.rs#L130-L131)

File: apps/docs-snippets/src/guide/cookbook/resubmitting-failed-transactions.test.ts [hunk 3]
```diff
@@ -70,8 +70,9 @@ describe('Resubmitting Failed Transactions', () => {
           await wallet.sendTransaction(transactionRequest);
         }
       }
+
```
Comment (petertonysmith94) [inline apps/docs-snippets/src/guide/cookbook/resubmitting-failed-transactions.test.ts:73]:
```suggestion
```

File: packages/fuel-gauge/src/funding-transaction.test.ts [hunk 4]
```diff
@@ -507,12 +507,10 @@ describe('Funding Transactions', () => {
...
-    await expectToThrowFuelError(
-      () => fundedWallet.transfer(receiver.address, transferAmount, provider.getBaseAssetId()),
-      new FuelError(
-        FuelError.CODES.INVALID_REQUEST,
-        'Transaction is not inserted. Hash is already known'
-      )
+    await expect(() =>
+      fundedWallet.transfer(receiver.address, transferAmount, provider.getBaseAssetId())
+    ).rejects.toThrowError(
+      /Transaction input validation failed: Transaction id already exists \(id: .*\)/
```
Comment (arboleya) [inline packages/fuel-gauge/src/funding-transaction.test.ts:513]: Shouldn't we validate if the error code is `FuelError.CODES.INVALID_REQUEST`?

File: packages/account/src/test-utils/launchNode.ts [hunk 5]
```diff
@@ -185,6 +185,7 @@ export const launchNode = async ({
         ['--ip', ipToUse],
         ['--port', portToUse],
         useInMemoryDb ? ['--db-type', 'in-memory'] : ['--db-path', tempDir],
+        ['--tx-ttl-check-interval', '1s'],
```
Comment (nedsalk) [inline packages/account/src/test-utils/launchNode.ts:188]: Why add this to the `launchNode` code? Shouldn't it be only added to `transaction-response.test.ts` tests as they were the ones failing?

File: packages/fuel-gauge/src/funding-transaction.test.ts (hunk 4 above)
Comment (petertonysmith94) [reply packages/fuel-gauge/src/funding-transaction.test.ts:513]: `expectToThrowFuelError` currently doesn't support this (from what I can tell).

Can implement this is required.

File: packages/account/src/test-utils/launchNode.ts (hunk 5 above)
Comment (petertonysmith94) [reply packages/account/src/test-utils/launchNode.ts:188]: Probably wise.

File: packages/fuel-gauge/src/funding-transaction.test.ts (hunk 4 above)
Comment (arboleya) [reply packages/fuel-gauge/src/funding-transaction.test.ts:513]: Pseuco-code.

```ts
const { error } = await safeExec(async () => {
  fundedWallet.transfer(receiver.address, transferAmount, provider.getBaseAssetId())
});

expect(error.code).toEqual(..)
expect(error.message).toEqual(..)
```

Comment (petertonysmith94) [reply packages/fuel-gauge/src/funding-transaction.test.ts:513]: https://github.com/FuelLabs/fuels-ts/pull/3272/commits/5aabc578a2b5c5b5a30ab2ca3998b06a68138db0

File: packages/account/src/test-utils/launchNode.ts (hunk 5 above)
Comment (petertonysmith94) [reply packages/account/src/test-utils/launchNode.ts]: https://github.com/FuelLabs/fuels-ts/pull/3272/commits/73094a1b91a3b28996024d57e470ea1dfdf9edf5

**Review by arboleya — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Review by danielbate — APPROVED** [review]

**Pull request merged**